aten/src/ATen/native/mkldnn/xpu/detail/QMatmul.cpp

```diff
@@ -328,4 +328,205 @@ void quantized_matmul(
+// logical dims
+inline ScaleSpec make_src_spec(ScalingType scaling_type, int64_t M, int64_t K) {
+  switch (scaling_type) {
+    case ScalingType::TensorWise:
+      return {0, {1, 1}, dnnl::memory::data_type::f32};
+    case ScalingType::RowWise:
+      return {(1 << 0) | (1 << 1), {1, K}, dnnl::memory::data_type::f32};
+  }
+  TORCH_CHECK(
+      false, "Unknown src scaling type, got ", static_cast<int>(scaling_type));
+}
+
+inline ScaleSpec make_wei_spec(ScalingType scaling_type, int64_t K, int64_t N) {
+  switch (scaling_type) {
+    case ScalingType::TensorWise:
+      return {0, {1, 1}, dnnl::memory::data_type::f32};
+    case ScalingType::RowWise:
+      return {(1 << 0) | (1 << 1), {K, 1}, dnnl::memory::data_type::f32};
+  }
+  TORCH_CHECK(
+      false, "Unknown wei scaling type, got ", static_cast<int>(scaling_type));
+}
+
+sycl::event scaled_matmul(
+    const Tensor& mat1,
+    const Tensor& mat2,
+    Tensor& result,
+    const Tensor& scale_a,
+    const Tensor& scale_b,
+    ScalingType scaling_choice_a,
+    ScalingType scaling_choice_b,
+    const std::optional<at::Tensor>& bias,
+    const std::optional<at::Tensor>& scale_result,
+    bool use_fast_accum) {
+  auto& engine = GpuEngineManager::Instance().get_engine();
+  auto& stream = GpuStreamManager::Instance().get_stream();
+
+  // This function will do steps with following steps
+  // 1. create memory descriptor
+  // 2. call write_to_dnnl_memory() to actually write memory
+  // 3. execute
+
+  const int64_t M = mat1.size(0);
+  const int64_t K = mat1.size(1);
+  const int64_t N = mat2.size(1);
+
+  // 1.1 Create memory descriptor
+  dnnl::memory::desc src_md = get_onednn_md(mat1);
+  dnnl::memory::desc weights_md = get_onednn_md(mat2);
+  dnnl::memory::desc dst_md = get_onednn_md(result);
+
+  // scale_a and scale_b has already be checked in `is_desired_scaling()` call.
+  // So we could directly get their memory desc and set later.
+  dnnl::memory::desc scale_a_md = get_onednn_md(scale_a);
+  dnnl::memory::desc scale_b_md = get_onednn_md(scale_b);
+
+  dnnl::memory::desc bias_md;
+  bool with_bias = bias.has_value();
+  at::Tensor possible_reshaped_bias = bias.value_or(at::Tensor());
+  if (with_bias) {
+    if (possible_reshaped_bias.dim() == 1) {
+      possible_reshaped_bias =
+          possible_reshaped_bias.reshape({1, possible_reshaped_bias.size(0)});
+      bias_md = get_onednn_md(possible_reshaped_bias);
+    } else {
+      bias_md = get_onednn_md(possible_reshaped_bias);
+    }
+  }
+
+  // 1.2 Create primitive descriptor and set scales mask
+  const ScaleSpec src_spec = make_src_spec(scaling_choice_a, M, K);
+  const ScaleSpec wei_spec = make_wei_spec(scaling_choice_b, K, N);
+
+  dnnl::primitive_attr op_attr = dnnl::primitive_attr();
+
+#if ONEDNN_SUPPORT_DETERMINISTIC
+  if (at::globalContext().deterministicAlgorithms() ||
+      at::globalContext().deterministicMkldnn())
+    op_attr.set_deterministic(true);
```

Comment:
Pls. check if oneDNN supports it. If not, please open a github issue to oneDNN repo.

Reply:
Yes, oneDNN supports deterministic mode.